internal/store/horizontalpodautoscaler.go

```diff
@@ -193,48 +186,41 @@ func hpaMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generat
 				ms := make([]*metric.Metric, 0, len(a.Status.CurrentMetrics))
 				for _, m := range a.Status.CurrentMetrics {
 					var metricName string
-
+					var currentMetric autoscaling.MetricValueStatus
 					// The variable maps the type of metric to the corresponding value
 					metricMap := make(map[metricTargetType]float64)
 
 					switch m.Type {
 					case autoscaling.ObjectMetricSourceType:
 						metricName = m.Object.Metric.Name
-
-						if m.Object.Current.Value != nil {
-							metricMap[value] = float64(m.Object.Current.Value.MilliValue()) / 1000
-						}
-						if m.Object.Current.AverageValue != nil {
-							metricMap[average] = float64(m.Object.Current.AverageValue.MilliValue()) / 1000
-						}
+						currentMetric = m.Object.Current
 					case autoscaling.PodsMetricSourceType:
 						metricName = m.Pods.Metric.Name
-
-						metricMap[average] = float64(m.Pods.Current.AverageValue.MilliValue()) / 1000
+						currentMetric = m.Pods.Current
 					case autoscaling.ResourceMetricSourceType:
 						metricName = string(m.Resource.Name)
-
-						if m.Resource.Current.AverageUtilization != nil {
-							metricMap[utilization] = float64(*m.Resource.Current.AverageUtilization)
-						}
-
-						if m.Resource.Current.AverageValue != nil {
-							metricMap[average] = float64(m.Resource.Current.AverageValue.MilliValue()) / 1000
-						}
+						currentMetric = m.Resource.Current
+					case autoscaling.ContainerResourceMetricSourceType:
+						metricName = string(m.ContainerResource.Name)
```

Comment:
this is already a string

Reply:
I am afraid of not. It's type corev1.ResourceName, like `case autoscaling.ResourceMetricSourceType`. Refer to https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/types.go#L296